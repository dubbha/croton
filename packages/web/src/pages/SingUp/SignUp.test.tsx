import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

describe('pages/SignUp', () => {
  const fn = jest.fn();

  it('should render successfully', () => {
    jest.isolateModules(() => {
      jest.doMock('react-redux', () => ({
        useStore: () => ({
          getState: () => ({}),
        }),
        useDispatch: () => fn,
      }));

      jest.doMock('components/SignUpForm', () => {
        const React = require('react');
        return {
          SignUpForm: ({ onSubmit }) => (
            <button onClick={onSubmit} data-testid="submit" />
          ),
        };
      });

      const { SignUp } = require('./SignUp');

      const { container } = render(<SignUp />, { wrapper: MemoryRouter });
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should dispatch on submit', () => {
    jest.isolateModules(() => {
      jest.doMock('react-redux', () => ({
        useStore: () => ({
          getState: () => ({}),
        }),
        useDispatch: () => fn,
      }));

      jest.doMock('components/SignUpForm', () => {
        const React = require('react');
        return {
          SignUpForm: ({ onSubmit }) => (
            <button onClick={onSubmit} data-testid="submit" />
          ),
        };
      });

      const { SignUp } = require('./SignUp');

      const { getByTestId } = render(<SignUp />, { wrapper: MemoryRouter });
      const submitButton = getByTestId('submit');

      fireEvent.click(submitButton);
      expect(fn).toBeCalled();
    });
  });
});
