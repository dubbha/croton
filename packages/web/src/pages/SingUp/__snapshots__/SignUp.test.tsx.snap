// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/SignUp should render successfully 1`] = `
<div
  class="container"
>
  <header>
    <div
      class="nav"
    >
      <div
        class="nav-item"
      >
        <a
          aria-current="page"
          class="nav-link active"
          href="/"
        >
          Home
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/signin"
        >
          Sign In
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/signup"
        >
          Sign Up
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/reset"
        >
          Reset Password
        </a>
      </div>
    </div>
  </header>
  <form
    class=""
  >
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="formFirstName"
      >
        First Name
      </label>
      <input
        class="form-control"
        data-testid="signUpForm__firstName"
        id="formFirstName"
        placeholder="Enter First Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="formLastName"
      >
        Last Name
      </label>
      <input
        class="form-control"
        data-testid="signUpForm__lastName"
        id="formLastName"
        placeholder="Enter Last Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="formEmail"
      >
        Email address
      </label>
      <input
        class="form-control"
        data-testid="signUpForm__email"
        id="formEmail"
        placeholder="Enter Email"
        type="email"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="formPassword"
      >
        Password
      </label>
      <input
        class="form-control"
        data-testid="signUpForm__password"
        id="formPassword"
        placeholder="Enter password"
        type="password"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="formPasswordMatch"
      >
        Repeat Password
      </label>
      <input
        class="form-control"
        data-testid="signUpForm__passwordMatch"
        id="formPasswordMatch"
        placeholder="Repeat password"
        type="password"
        value=""
      />
    </div>
    <button
      class="btn btn-primary"
      data-testid="submitButton"
      disabled=""
      type="submit"
    >
      Sing Up
    </button>
  </form>
  <div
    class="signup-login"
  >
    Already have an account ?
    <a
      href="/signin"
    >
       Sign In
    </a>
  </div>
  <footer>
    <button
      class="btn btn-light"
      type="button"
    >
      Footer
    </button>
  </footer>
</div>
`;
